pathplannerlib/src/main/java/com/pathplanner/lib/config/RobotConfig.java

```diff
@@ -304,4 +306,171 @@ public static RobotConfig fromGUISettings() throws IOException, ParseException {
       return new RobotConfig(massKG, MOI, moduleConfig, trackwidth);
     }
   }
+
+  // Validation alerts
+  private static final Alert BAD_GUI_CONFIG =
+      new Alert("GUI Config Couldn't be loaded", AlertType.kError);
+  private static final Alert MOI_ALERT = new Alert("MOI Config Mismatch", AlertType.kError);
+  private static final Alert MASS_ALERT = new Alert("Mass Config Mismatch", AlertType.kError);
+  private static final Alert TORQUE_ALERT = new Alert("Torque Friction Mismatch", AlertType.kError);
+  private static final Alert CURRENT_ALERT =
+      new Alert("Drive Current Limit Mismatch", AlertType.kError);
+  private static final Alert MOTOR_ALERT =
+      new Alert("Drive Motor Config Mismatch", AlertType.kError);
+  private static final Alert VELOCITY_ALERT =
+      new Alert("Max Drive Velocity Mismatch", AlertType.kError);
+  private static final Alert COF_ALERT = new Alert("Wheel COF Mismatch", AlertType.kError);
+  private static final Alert RADIUS_ALERT = new Alert("Wheel Radius Mismatch", AlertType.kError);
+  private static final Alert LOCATION_ALERT =
+      new Alert("Module Location Mismatch", AlertType.kError);
+
+  /**
+   * Checks if this configuration matches the GUI configuration. Loads the GUI config and compares
+   * all properties, setting alerts for any mismatches.
+   *
+   * @return true if all configuration matches GUI config, false if any are invalid or GUI config
+   *     cannot be loaded
+   */
+  public boolean hasValidConfig() {
+    RobotConfig guiConfig;
+    try {
+      guiConfig = RobotConfig.fromGUISettings();
+    } catch (IOException | ParseException e) {
+      BAD_GUI_CONFIG.set(true);
+      return false;
+    }
+
+    return validatePhysicalProperties(guiConfig)
+        && validateDriveSystem(guiConfig)
+        && validateWheelProperties(guiConfig)
+        && validateModuleLocations(guiConfig);
```

Comment:
If one of these methods fails, all methods afterwards would not be run, meaning that alerts that should be set might not actually get set.